pyomo/core/base/objective.py

```diff
@@ -427,6 +414,22 @@ def __init__(self, *args, **kwd):
     # construction
     #
 
+    def __call__(self, exception=True):
+        if self._constructed:
+            if len(self._data) == 0:
+                raise ValueError(
+                    "Evaluating the expression of ScalarObjective "
+                    "'%s' before the Objective has been assigned "
+                    "a sense or expression. There is currently "
```

Comment:
There is a minor inconsistency across error messages ("There is currently nothing to access" vs. "(there is currently no value to return)") Can you settle on one format?